### PR TITLE
[ASCII-2055] Add new e2e test for agent hostname on GCP

### PR DESCRIPTION
**Pull request by pgimalac**
### What does this PR do?
Add an e2e test for the hostname subcommand on GCP.

### Motivation
Ensuring hostname logic works as expected on GCP cloud provider.

### Describe how to test/QA your changes
N/A

### CODE REVIEW

File: .gitlab/e2e/e2e.yml [hunk 1]
```diff
@@ -139,7 +139,7 @@ new-e2e-agent-subcommands:
       - EXTRA_PARAMS: --run "Test(Linux|Windows)StatusSuite"
       - EXTRA_PARAMS: --run "Test(Linux|Windows)HealthSuite"
       - EXTRA_PARAMS: --run "Test(Linux|Windows)ConfigSuite"
-      - EXTRA_PARAMS: --run "Test(Linux|Windows|LinuxAzure)HostnameSuite"
+      - EXTRA_PARAMS: --run "Test(Linux|Windows|LinuxAzure|LinuxGCP)HostnameSuite"
```
Comment (GustavoCaso) [inline .gitlab/e2e/e2e.yml:142]: What about making the regex take all Linux tests with `Linux.*`? 
`--run "Test(Linux.*|Windows)HostnameSuite"`

**Review by GustavoCaso — APPROVED** [review]
LGTM

**Review by ogaca-dd — APPROVED** [review]

File: .gitlab/e2e/e2e.yml (hunk 1 above)
Comment (pgimalac) [reply .gitlab/e2e/e2e.yml:142]: I like the fact that the suites are explicitly listed, you can't accidentally run something, while if you start introducing `.*` you could run more than intended...

**Review by KevinFairise2 — APPROVED** [review]

**Pull request merged**